proposals/nnnn-enhance-video-streaming-performance-for-android.md

```diff
@@ -0,0 +1,115 @@
+# Enhance Video Streaming Performance for Android
+
+* Proposal: [SDL-NNNN](nnnn-enhance-video-streaming-performance-for-android.md)
+* Author: [BrandonHe](https://github.com/BrandonHe)
+* Status: **Awaiting review**
+* Impacted Platforms: [Android]
+
+## Introduction
+
+This proposal aims to enhance SDL Android library for massive data transmission performance for video streaming and projection by refactor default I/O pipe write/read class.
+
+## Motivation
+
+The SDL Android library are currently using default java.io package, it uses `PipedInputStream` via `write()` function to stream data **Byte by Byte**, because CPU operation is very fast, in most cases, the operation performance as good as we expected. But for video streaming and projection, it is a botterneck for transmission according to test. It will cause very frequently CPU resources switch between read thread and write thread. Thus has an effect on video streaming performance. Another motivation here is that if read thread get the critical resource to read after write thread just write one byte or serveral bytes, this leads to slice one frame data to send by many times. It lso reduces the efficiency of data transmission and increase the CPU workload.
```

Comment:
Please make the following changes in this paragraph:

- Change "performance" to "performs"
- Change "botterneck" to "bottleneck" 
- Change "frequently" to "frequent"
- Change "lso" to "also"